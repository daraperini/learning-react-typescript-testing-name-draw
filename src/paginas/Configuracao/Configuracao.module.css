.titulo {
  font-size: 20px;
  font-weight: 600;
  color: #4b69fd;
}

.container__rodape {
  display: flex;
  flex-direction: column;
  gap: 10rem;
  align-items: center;
}

@media screen and (min-width: 799px) {
  .titulo {
    font-size: 32px;
  }

  .container__rodape {
    flex-direction: row;
    position: absolute;
    bottom: 15%;
  }
}
