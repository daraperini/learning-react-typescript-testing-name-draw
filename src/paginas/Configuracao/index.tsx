import Formulario from "../../componentes/Formulario";
import ListaParticipantes from "../../componentes/ListaParticipantes";
import Rodape from "../../componentes/Rodape";
import imagem from "./imagens/sacolas.png";
import styles from "./Configuracao.module.css";

const Configuracao = () => {
  return (
    <>
      <h1 className={styles.titulo}>Vamos come√ßar!</h1>
      <Formulario />
      <ListaParticipantes />
      <div className={styles.container__rodape}>
        <Rodape />
        <img src={imagem} alt="Duas sacolas sobrepostas" />
      </div>
    </>
  );
};

export default Configuracao;
