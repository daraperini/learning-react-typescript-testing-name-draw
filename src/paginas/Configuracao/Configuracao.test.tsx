import React from "react";
import { RecoilRoot } from "recoil";
import Configuracao from ".";
import { render } from "@testing-library/react";

const mockNavegacao = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavegacao,
  };
});

describe("a página de configuração", () => {
  test("deve ser renderizada corretamente", () => {

    // pega o container onde foi renderizado
    const {container} = render(
      <RecoilRoot>
        <Configuracao />
      </RecoilRoot>
    )

    // espera que todas as renderizações sejam iguais visualmente
    expect(container).toMatchSnapshot()
  });
});
