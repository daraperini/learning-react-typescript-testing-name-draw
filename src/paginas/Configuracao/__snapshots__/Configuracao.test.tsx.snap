// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <h1
    class="titulo"
  >
    Vamos começar!
  </h1>
  <form
    class="formulario"
  >
    <input
      class="input"
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      class="botao"
      disabled=""
    >
      Adicionar
    </button>
    
  </form>
  <ul
    class="lista"
  />
  <div
    class="container__rodape"
  >
    <footer>
      <button
        class="botao"
        disabled=""
      >
        Iniciar brincadeira!
      </button>
    </footer>
    <img
      alt="Duas sacolas sobrepostas"
      src="sacolas.png"
    />
  </div>
</div>
`;
